src/snyk/base/modules/baseSnykModule.ts

```diff
@@ -43,12 +48,16 @@ export default abstract class BaseSnykModule implements IBaseSnykModule {
       configuration,
       this.openerService,
       this.viewManagerService,
-      this.filesWatcher,
+      vsCodeWorkspace,
       this.processError.bind(this),
     );
+    this.scanModeService = new ScanModeService(this.contextService);
+    this.notificationService = new NotificationService(vsCodeWindow);
   }
 
   abstract processError(error: errorType, options?: { [key: string]: any }): Promise<void>;
 
-  abstract startExtension(): Promise<void>;
+  abstract runScan(): Promise<void>;
```

Comment:
runScan() will use runCodeScan() and runOssScan() or it have some different purpose?

Reply:
Yes, it will run both of these functions and check authentication on top of it. You can see the logic in snyk/base/modules/snyklib.ts